examples/gcoap/client.c

```diff
@@ -141,18 +141,48 @@ static void _resp_handler(const gcoap_request_memo_t *memo, coap_pkt_t* pdu,
     }
 }
 
-static bool _parse_endpoint(sock_udp_ep_t *remote,
-                            const char *addr_str, const char *port_str)
+static bool _parse_endpoint(sock_udp_ep_t *remote, const char *addr_str, const char *port_str)
```

Comment:
Can't this entire function be replaced by `sock_udp_name2ep()` ?

Reply:
Only if we change the user input scheme for the shell from `coap proxy set fe80::d33f:536a:9c15:2af4 5684` to `coap proxy set [fe80::d33f:536a:9c15:2af4]:5684`. That would require a few changes in argument handling of the `gcoap` example. Do you think it would be worth it to unify the codebase? I could do it.

Reply:
Yes please! The inconsistency of the gcoap example CLI had annoyed me for a while now

Reply:
See #17870